chain/cosmos/module_authz.go

```diff
@@ -0,0 +1,129 @@
+package cosmos
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+)
+
+// AuthzGrant grants a message as a permission to an account.
+func (tn *ChainNode) AuthzGrant(ctx context.Context, granter ibc.Wallet, grantee, authType string, extraFlags ...string) (*sdk.TxResponse, error) {
+
+	allowed := "send|generic|delegate|unbond|redelegate"
+	if !strings.Contains(allowed, authType) {
+		return nil, fmt.Errorf("invalid auth type: %s allowed: %s", authType, allowed)
+	}
+
+	cmd := []string{"authz", "grant", grantee, authType}
+
+	// when using the generic type, you must specify a --msg-type flag
+	if authType == "generic" {
+		msgTypeIndex := -1
+		for i, flag := range extraFlags {
+			if flag == "--msg-type" {
+				msgTypeIndex = i
+				break
+			}
+		}
+
+		if msgTypeIndex == -1 {
+			return nil, fmt.Errorf("missing --msg-type flag when granting generic authz")
+		}
+
+		msgType := extraFlags[msgTypeIndex+1]
+		if !strings.HasPrefix(msgType, "/") {
+			extraFlags[msgTypeIndex+1] = "/" + msgType
+		}
+	}
+
+	cmd = append(cmd, extraFlags...)
+
+	txHash, err := tn.ExecTx(ctx, granter.KeyName(),
+		append(cmd, "--output", "json")...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return tn.TxHashToResponse(ctx, txHash)
+}
+
+// AuthzExec executes an authz MsgExec transaction with a single nested message.
+func (tn *ChainNode) AuthzExec(ctx context.Context, grantee ibc.Wallet, nestedMsgCmd []string) (*sdk.TxResponse, error) {
+	fileName := "authz.json"
+	if err := createAuthzJSON(ctx, tn, fileName, nestedMsgCmd); err != nil {
+		return nil, err
+	}
+
+	txHash, err := tn.ExecTx(ctx, grantee.KeyName(),
+		"authz", "exec", tn.HomeDir()+"/"+fileName,
```

Comment:
in our case the underlying OS is not likely to change but using `filepath.Join` would ensure the path modification is happening in an OS agnostic manner.

Reply:
https://github.com/strangelove-ventures/interchaintest/pull/892/commits/795395957e46cab67989d87eca78197c6fd13f1a